xsnap/documentation/xsnap-worker.md

```diff
@@ -0,0 +1,52 @@
+# xsnap-worker
+
+`sources/xsnap-worker.c` contains a variant of `xsnap` which accepts execution commands over a file descriptor. It is designed to function as a "JS coprocessor", driven by a parent process (which can be written in any language). The parent does a fork+exec of `xsnap-worker`, then writes netstring-formatted commands to the child. The child executes those commands (evaluating JS or delivering the command to a handler function), possibly emitting one or more requests to the parent during execution, then finally finishes the command and writing a status to the parent (including metering information).
+
+By default, the process starts from an empty JS environment (not even SES or `lockdown`). If the child is started with a `-r SNAPSHOTFILENAME` argument, it will start from a previously-written heap snapshot instead.
+
+The launch arguments are:
+
+* `-h`: print this help message
+* `-i <interval>`: set the metering interval (TODO: what does this mean?)
```

Comment:
The details are tricky (and likely to evolve). The idea is straightforward. Checking the meter for overflow takes some time. The interval is a way to check less frequently. A larger interval means code is likely to exceed its computron budget by more, but will execute more quickly.